### ISSUE TITLE
org-journal-new-entry clears the previous day's file

### ISSUE THREAD

**Issue by akirak** [bastibe/org-journal#102]
Hi,

org-journal worked as expected until yesterday, but it doesn't now. Whenever I try to create a new entry using `org-journal-new-entry`, it opens the last day's file, make it empty, and saves the file. Fortunately, it wasn't fatal because the file was in Git so I could recover it, but I can't create a new entry for now.

My configuration is as follows. Do you have any idea what's gone wrong?

```emacs-lisp
(use-package org-journal
  :when (file-directory-p "~/personal/org-journal/")
  :config
  (defun org-journal-find-location ()
    "Go to the beginning of the today's journal file.

This can be used for an org-capture template to create an entry in the journal."
    ;; Open today's journal, but specify a non-nil prefix argument in order to
    ;; inhibit inserting the heading; org-capture will insert the heading.
    (org-journal-new-entry t)
    ;; Position point on the journal's top-level heading so that org-capture
    ;; will add the new entry as a child entry.
    (goto-char (point-min)))
  :custom
  (org-extend-today-until 4)
  (org-journal-date-format "%F (%a)")
  (org-journal-enable-agenda-integration t "Add future journal files to agenda")
  (org-journal-carryover-items t "Carry over unfinished tasks in the past")
  (org-journal-dir "~/personal/org-journal/"))
```

**Comment by bastibe** [comment]
Two days ago, I added support for `org-extend-today-until`. This is probably what introduced the problem for you. I don't understand how it would do that, though, and I can't reproduce it on my end.

But I did notice that you set your `org-journal-carryover-items` to `t`. This should carry over *every* headline. This rings familiar since you claim that a file is emptied completely. Is this intentional?
